jep/0000/README.adoc

```diff
@@ -0,0 +1,127 @@
+
+== Motivation
+
+Currently it is not possible to run Jenkins in a link:https://csrc.nist.gov/pubs/fips/140-2/upd2/final[FIPS-140-2] compliant way.  Many users needing Jenkins are required (e.g. for US government projects) to have any solution be compliant with FIPS-140 and today it is not possible for them to do this. 
+
+Both Jenkins core and plugins use cryptography that is not allowed and/or are not from an approved implementation, and this prevents adoption of Jenkins by these end users.
+Examples would be during setup where the install wizard creates a first admin user using the internal security realm which uses ``bcrypt`` which would fall foul of the standard due to its use of ``blowfish`` which is not an approved algorithm.
+
+== Reasoning
+
+To be FIPS-140 compliant, Jenkins must use validated, FIPS-140 compliant encryption algorithms. 
+This requires complex configuration of the host, the Java Virtual Machine (JVM), the servlet environment, and the Jenkins controller. 
+Changes may be required in both Jenkins core and plugins.
+As such it is not considered that the Jenkins community supports or provides instructions on how to run Jenkins in a compliant manner, but merely states that FIPS-140 compliant operation may be  possible with appropriate configuration of the JVM, Servlet environment, limited set of plugins, and finally instructing Jenkins to prefer a FIPS-140 mode of operation.
+
+As of the time of writing the validated Java implementations are only FIPS-140-2 compliant even though FIPS-140-3 has superseded it.  As such the initial implementations will be targeting the older FIPS-140-2 standard.
+
+== Backwards Compatibility
+
+It is out of scope to support previous configuration, as it would not be possible to migrate non FIPS-140 compliant data to FIPS-140 compliant data within a FIPS-140 compliant environment.  As such there is no backwards compatibility with previous versions.
+
+It may but is not a requirement of this JEP be possible to switch from a FIPS-140 compliant environment to a non FIPS-140 compliant environment.
+
+Code changes will however be backwards compatible; if the system is running in a non FIPS-140 compliant environment then it will continue to run and operate the same after any changes due to this JEP.
+
+== Security
+
+Some of the cryptography supported in FIPS 140 (especially FIPS 140-2) is considered less secure than some modern alternatives.  FIPS 140-2 cryptography will be used when Jenkins is running in FIPS-140 mode even if that cryptography is considered less secure than other alternatives.
+
+== Infrastructure Requirements
+
+There are no new infrastructure requirements related to this proposal.
+
+== Testing
+
+New code that changes behaviour when it detects the desire to run in FIPS-140 mode would be covered by unit tests like any other code change.  There would be no new testing introduced in a FIPS-140 compliant environment due to the complexity of maintaining this setup.
```

Comment:
> There would be no new testing introduced in a FIPS-140 compliant environment due to the complexity of maintaining this setup.

An assertion made without evidence. Why is it true?

Even if maintaining an automated test setup is too complex, the JEP template asks that authors describe their manual test plan. Why is that not done in this JEP?

Reply:
I will create a PR to update this section

Reply:
⇨ #404